src/main/java/com/google/devtools/build/lib/starlarkbuildapi/python/PyRuntimeInfoApi.java

```diff
@@ -216,6 +236,7 @@ PyRuntimeInfoApi<?> constructor(
         Object coverageFilesUncast,
         String pythonVersion,
         String stubShebang,
+        Object bootstrapTemplate,
```

Comment:
Same comment as above -- this should be Artifact instead of Object

Reply:
Is Artifact a starlark-native type? Even `interpreterUncast` is an Object here even though it's an Artifact

Reply:
interpreter is probably set as Object because it's input types are FileApi and None (i.e. Starlark.NONE), so there's no single Java type to capture both types in this function signature.

Artifact is starlark native (it can transparently go between java/starlark layers without any special conversion). It shows up as a File in starlark (Artifact implements FileApi)

Oh, I see, this is one of those "api interface" classes. Ok, yeah, FileApi is probably the more appropiate type, then. Looking at JavaInfoApi, that's what it does (and then just blinding casts to Artifact within the actual implementation). So I guess just follow that pattern.

I don't want to be too picky here because this code is slated for removal soon. So long as we don't end up with a Java stack trace on bad input, and bad inputs are rejected, I'm happy.